transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -205,15 +205,19 @@ public void run() {
     }
 
     private void notifyConnect(ChannelFuture future, Promise<Channel> promise) throws Exception {
```

Comment:
Good catch!

The issue is that this method is called from a future listener as well as directly from some methods. Only when this is called from a future listener, the `promise` is left incomplete. If you look for this pattern, there is another method which has the same issue: `releaseAndOfferIfHealthy()`. Can you also fix that method like you did here?

Reply:
For `releaseAndOfferIfHealthy` only the one inside the `listener` is required. The other above is caught by `doReleaseChannel`

Reply:
That is the same for the `notifyConnect` usage too.

Reply:
* Methods that are executed in listeners have now their own try/catch.
* I moved the asserts into try/catch because if I am running my tests, I wish to fail the promise instead of keeping my tests running forever. The same for `checkNotNull`
* Reviewed also `FixedChannelPool`